frontend/src/components/search/SearchResults.js

```diff
@@ -0,0 +1,287 @@
+
+const CantFindIt = () => (
+  <Row type="flex" justify="center">
+    <Col sm={22} md={18} lg={8} className="padding-large--top padding-xxl--bottom">
+      <h2>Can&#39;t find it?</h2>
+      <p>
+        Try clicking to see more results, or select the advanced lookup if available. Also, double
+        check your spelling to ensure it is correct. If you feel there is a problem, contact the
+        Core administrator to ask for assistance.
+      </p>
+    </Col>
+  </Row>
+);
+
+export class SearchResults extends Component {
+  state = {
+    isSearching: false,
+    hasSearchTerm: false,
+    params: {},
+  };
+
+  componentDidMount = () => {
+    if (!this.props.partyRelationshipTypeHash.PMT) {
+      this.props.fetchPartyRelationshipTypes();
+    }
+    if (!this.props.searchOptions.length) {
+      this.props.fetchSearchOptions();
+    }
+    this.handleSearch(this.props.location);
+  };
+
+  componentWillReceiveProps = (nextProps) => {
+    const locationChanged = nextProps.location !== this.props.location;
+    if (locationChanged) {
+      this.handleSearch(nextProps.location);
+    }
+
+    // The intention below is to trigger every time the object is updated, even if the
+    // array contents are the same, therefore compare with != vs !==
+    // eslint-disable-next-line eqeqeq
+    if (nextProps.searchResults != this.props.searchResults) {
+      this.setState({ isSearching: false });
+    }
+  };
+
+  handleSearch = (location) => {
+    const parsedParams = queryString.parse(location.search);
+    const { q, t } = parsedParams;
+
+    if (q) {
+      this.props.fetchSearchResults(q, t);
+      this.setState({ params: parsedParams, isSearching: true, hasSearchTerm: true });
+    }
+  };
+
+  render = () => {
+    const groupedSearchResults = [];
+    Object.entries(this.props.searchResults).forEach((entry) => {
+      const resultGroup = {
+        type: entry[0],
+        score: sumBy(entry[1], "score"),
+        results: map(entry[1], "result"),
+      };
+      if (resultGroup.score > 0) groupedSearchResults.push(resultGroup);
+    });
+
+    groupedSearchResults.sort((a, b) => a.score - b.score);
+    groupedSearchResults.reverse();
+
+    const type_filter = this.state.params.t;
+
+    if (this.state.isSearching) return <Loading />;
+
+    const results = this.props.searchTerms.map((t) => `"${t}"`).join(", ");
+
+    return (
+      this.state.hasSearchTerm && (
+        <div className="landing-page">
+          <div>
+            <div className="landing-page__header">
+              <h1 className="padding-small--bottom">
+                {`${
+                  type_filter ? this.props.searchOptionsHash[type_filter] : "Search results"
+                } for ${results}`}
+              </h1>
+              <div>
+                {type_filter ? (
+                  <LinkButton
+                    key="all"
+                    onClick={() => {
+                      this.props.history.push(`/search?q=${this.state.params.q}`);
```

Comment:
We don't use `this.props.history.push()` anywhere else in the app, 
for consistency, the route should have a dynamic route that we pass the param into

```
export const SEARCH_RESULTS = {
  route: "/search",
dynamicRoute: (q, t) => t ? `/search?q={q}&t={t}` : `/search?q={q}`
  component: SearchResults,
};
```

Reply:
this.props.history.push is used in a few places, although in conjunction with a dynamic route. I'll update it to be in line with the others.

Reply:
Updated 👍